src/Assigning/Assignment.hs

```diff
@@ -361,20 +342,18 @@ instance (Enum grammar, Eq1 ast, Ix grammar, Show grammar) => Parsing (Assignmen
   (<?>) :: HasCallStack => Assignment ast grammar a -> String -> Assignment ast grammar a
   a <?> s = tracing (Label a s) `Then` pure
 
-  unexpected :: HasCallStack => String -> Assignment ast grammar a
+  unexpected :: String -> Assignment ast grammar a
```

Comment:
Thought: instead of just ditching this constraint, can we glom it on to the passed `String`? I feel like call stacks are good and we shouldn’t lose these constraints, but maybe I’m bugging.

Reply:
I’ve only removed redundant constraints. This call stack was never used.

Reply:
Fair enough. I guess it would still be preserved if we called `error` here? But we don’t, so no matter.

Reply:
Yes, it wouldn’t have been redundant in that case, and I wouldn’t have removed it 😁